File: 22 Dec 960. Delete Columns to Make Sorted III/code.cpp
class Solution {
public:
    int minDeletionSize(vector<string>& strs) {
        int n=strs.size(),m=strs[0].size();
        vector<int>dp(m,1);
        // if(n==1)return m-1;
        for(int i=1;i<m;i++){
            for(int j=0;j<i;j++){
                int f=1;
                for(auto c:strs){
                    if(c[j]>c[i]){f=0;break;}
                }
                if(f)dp[i]=max(dp[i],dp[j]+1);
            }
        }
        // for(int i:dp)cout<<i<<" ";
        return m-*max_element(begin(dp),end(dp));
    }
};

File: 13 Jan 2026 3453. Separate Squares I/code.py
class Solution:
    def separateSquares(self, squares: List[List[int]]) -> float:
        l,r=0,max(y+z for x,y,z in squares)
        res=0
        while(abs(l-r)>(10**(-5))):
            m=(l+r)/2
            fa,sa=0,0
            for x,y,h in squares:
                if(y>=m):
                    fa+=(h*h)
                elif ((y+h)<m):
                    sa+=(h*h)
                else:
                    fa+=((y+h-m)*h)
                    sa+=(m-y)*h
            if(fa>sa):
                l=m
            else:
                r=m
        return r